webapp/src/main/resources/public/js/components/jobs/CronExpressionInput.jsx

```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { DropdownButton, MenuItem } from "react-bootstrap";
+import cronstrue from "cronstrue";
+import { validateCronExpression } from "../../utils/CronExpressionHelper";
+
+const CRON_EXPRESSION_OPTIONS = [
+    { display: "Every 5 Minutes", cron: "0 0/5 * * * ?" },
+    { display: "Every Hour", cron: "0 0 * * * ?" },
+    { display: "Every Day at Midnight (UTC)", cron: "0 0 0 * * ?" },
+    { display: "Custom Cron Expression", cron: "__custom__" },
+];
+
+class CronExpressionInput extends React.Component {
+    static propTypes = {
+        cron: PropTypes.string,
+        onChange: PropTypes.func.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+        const matchedOption = CRON_EXPRESSION_OPTIONS.find(option => option.cron === props.cron);
+        this.state = {
+            selectedOption: props.cron
+                ? (matchedOption ? matchedOption.cron : "__custom__")
+                : null,
+            customCron: matchedOption || !props.cron ? "" : (props.cron || "")
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.cron !== this.props.cron) {
+            const matchedOption = CRON_EXPRESSION_OPTIONS.find(option => option.cron === this.props.cron);
+            if (this.state.selectedOption !== "__custom__" || (this.props.cron && matchedOption)) {
+                this.setState({
+                    selectedOption: this.props.cron
+                        ? (matchedOption ? matchedOption.cron : "__custom__")
+                        : null,
+                    customCron: matchedOption || !this.props.cron ? "" : (this.props.cron || "")
+                });
+            }
+        }
+    }
+
+    handleDropdownSelect = (eventKey) => {
```

Comment:
Nit: can we rename this? 
`eventKey` usually implies a single key pressed but in this case it's the full text from the looks of it